File: app-java/src/main/java/com/tritiumstudios/phasmophobiaevidencetool/android/activities/mainmenus/newsletter/data/NewsletterMessageData.java
package com.tritiumstudios.phasmophobiaevidencetool.android.activities.mainmenus.newsletter.data;

import com.tritiumstudios.phasmophobiaevidencetool.android.data.utilities.FontUtils;

public class NewsletterMessageData {

    private String title, description, date;

    public NewsletterMessageData(String title, String description, String date) {
        setTitle(FontUtils.removeXMLImgSrcTags(title));
        setDescription(FontUtils.removeXMLImgSrcTags(description));
        setDate(FontUtils.removeXMLPubDateClockTime(FontUtils.removeXMLPubDateClockTime(date)));
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getDate() {
        return date;
    }

    public boolean hasContent() {
        return hasTitle() || hasDescription() || hasDate();
    }

    private boolean hasDate() {
        return getDate() != null && !getDate().equals("");
    }

    private boolean hasDescription() {
        return getDescription() != null && !getDescription().equals("");
    }

    private boolean hasTitle() {
        return getTitle() != null && !getTitle().equals("");
    }


}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/mainmenus/newsletter/data/NewsletterMessageData.kt
package com.TritiumGaming.phasmophobiaevidencepicker.activities.mainmenus.newsletter.data

import com.TritiumGaming.phasmophobiaevidencepicker.data.utilities.FontUtils

class NewsletterMessageData(title: String?, description: String?, date: String?) {
    @JvmField
    var title: String? = null
    @JvmField
    var description: String? = null
    @JvmField
    var date: String? = null

    init {
        this.title = FontUtils.removeXMLImgSrcTags(title)
        this.description = FontUtils.removeXMLImgSrcTags(description)
        this.date = FontUtils.removeXMLPubDateClockTime(FontUtils.removeXMLPubDateClockTime(date))
    }

    fun hasContent(): Boolean {
        return hasTitle() || hasDescription() || hasDate()
    }

    private fun hasDate(): Boolean {
        return date != null && !date!!.isEmpty()
    }

    private fun hasDescription(): Boolean {
        return description != null && !description!!.isEmpty()
    }

    private fun hasTitle(): Boolean {
        return title != null && !title!!.isEmpty()
    }
}
